### PR TITLE
Fix unused params warnings plus incomplete struct init (done to default values).

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

**Comment by JohnLangford** [comment]
This pull request **introduces 1 alert** when merging 41c3c0f8b1b8e58441535bb3abab67122abf5137 into 2b64c3c0938d1f03aa8e6c9ca1513174a14240e2 - [view on LGTM.com](https://lgtm.com/projects/g/VowpalWabbit/vowpal_wabbit/rev/pr-436f7399a8f9f735e34a82dd39746416c7e4b640)

**new alerts:**

* 1 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by JohnLangford** [comment]
This pull request **introduces 1 alert** when merging 0d835d276e68d769cba3e0119ba405e8506e5f7c into 2b64c3c0938d1f03aa8e6c9ca1513174a14240e2 - [view on LGTM.com](https://lgtm.com/projects/g/VowpalWabbit/vowpal_wabbit/rev/pr-892996052ab50814999c2c72206150c9ec88d540)

**new alerts:**

* 1 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by JohnLangford** [comment]
What is the fixme comment?

File: vowpalwabbit/lda_core.cc [hunk 1]
```diff
@@ -85,7 +85,7 @@ struct lda
 // if it makes it into VS 2015 change the next ifdef to check Visual Studio Release
 
   // static constexpr float underflow_threshold = 1.0e-10f;
-  inline const float  underflow_threshold() { return 1.0e-10f; }
+  inline float  underflow_threshold() { return 1.0e-10f; }
```
Comment (JohnLangford) [inline vowpalwabbit/lda_core.cc]: What's going on with const here?

File: vowpalwabbit/parser.cc [hunk 2]
```diff
@@ -484,6 +484,7 @@ void enable_sources(vw& all, bool quiet, size_t passes)
     // background process (if foreground is not set)
     if (!all.opts_n_args.vm.count("foreground"))
     {
+      //FIXME switch to posix_spawn
```
Comment (JohnLangford) [inline vowpalwabbit/parser.cc]: I'm unfamiliar with posix_spawn.  What's needed?

**Comment by JohnLangford** [comment]
This pull request **introduces 1 alert** when merging dadd8b7bcb841e8d3f785c8c32c93ef893d4e3b7 into 6dbb7e5f96e701253f16397b19ecde70881a7057 - [view on LGTM.com](https://lgtm.com/projects/g/VowpalWabbit/vowpal_wabbit/rev/pr-89a1785971448f211f5cb8f3c1236cbbb2de652d)

**new alerts:**

* 1 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: vowpalwabbit/parser.cc (hunk 2 above)
Comment (jackgerrits) [reply vowpalwabbit/parser.cc]: Could we use a std::thread instead so that the code is portable?

File: vowpalwabbit/lda_core.cc (hunk 1 above)
Comment (kumpera) [reply vowpalwabbit/lda_core.cc]: const is semantically meaningless when returning POD values as they get copied.

File: vowpalwabbit/parser.cc (hunk 2 above)
Comment (kumpera) [reply vowpalwabbit/parser.cc]: @JohnLangford daemon is deprecated on OSX. 
Additionally posix_spawn gives fine grained control over the signal mask and all the process IDs.

It's not clear to me if those are features needed by VW particular.

@jackgerrits both daemon and posix_spawn create a new process and the former exits the parent. Threads can't provide that.

**Comment by JohnLangford** [comment]
Merged, thanks :-)

**Pull request merged**